serialize.c

```diff
@@ -0,0 +1,111 @@
+#include "serialize.h"
+
+char binary_to_base64(uint8_t binary) {
+    if(binary <= 25) return binary + 'A';
+    if(binary <= 51) return (binary - 26) + 'a';
+    if(binary <= 61) return (binary - 52) + '0';
+    if(binary == 62) return '&';
+    return '/';
```

Comment:
Can you add a check for if binary is greater than 63, and is thus illegal? I don't know how to handle that error currently, but we shouldn't just silently return `'/'` as we do here

Reply:
What should the code do if binary is greater than 63?

Reply:
In a different branch we have a report_error function that will be called whenever a programming error like that shows up. Once both branches are merged to master we’ll call that, for now just leave a comment reminding us to report the problem. And return 0 or something

Reply:
Fixed in 049a07c.